wordpress-mobile/WordPress-FluxC-Android
# Woo REST API: media upload using application passwords

Closes https://github.com/woocommerce/woocommerce-android/issues/8017

This PR adds a media rest client that supports application passwords, I'll explain some decisions in the code below.

tip: reviewing commit by commit might make understanding the changes easier.

### Testing
1. Open the example app.
2. Sign in using site credentials.
3. Open media screen.
4. Test some scenarios (make sure the app has the storage permission, Android removes it sometimes).
5. Confirm the scenarios work as expected.
